internal/bed/read.go

```diff
@@ -0,0 +1,84 @@
+package bed
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Opening and reading the bed file
+func (bf *Bedfile) Read() error {
+	file, err := os.Open(bf.Input)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	return bf.read(file)
+}
+
+// Reading the bed file
+func (bf *Bedfile) read(file io.Reader) error {
+	var err error
+	var expectedNrOfCols int
+
+	minNrCols := 3
+	headerPattern := regexp.MustCompile(`^(browser|track|#)`)
+	strandPattern := regexp.MustCompile(`^(\.|\+|-|\+1|-1|1)$`)
+
+	lineNr := 0
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var l Line
+		lineNr++
+
+		lineText := scanner.Text()
+
+		// Handle headers
+		if headerPattern.MatchString(lineText) && len(bf.Lines) == 0 {
+			bf.Header = append(bf.Header, lineText)
+			continue
+		}
```

Comment:
It can, and after the header the number of columns might change.

After a lot of reading I have figured out that the files with headers are actually referred to as track files and not bed files, so I have decided not add more support for track files at the moment. If there should be header elements further down in the file, the user will get an error telling them that there is an unexpected number of columns for the line containing the header. The second last test case in TestRead() are actually testing that we get the correct error if this should be the case :)

Reply:
cool, thanks for your explanation 👍

Reply:
So, as I understand the track files would fail in line 56 because they have a different column count, right? Could it happen that a track file has the same column count as the bed entries meaning could it happen that it doesn't fail in 56?

Reply:
It would fail either way as the header would usually constitute one column as it does not contains any `\t`, and a bed file needs to contain a minimum of three columns :)

Reply:
Just to clarify: We record the number of column for the first non-header line and check that all subsequent lines contains the same number of columns. So because of the bed file standard the number of columns will always be >= 3, and since the header does usually constitute one line without any `\t` this error should be picked up.